input_definition.go

```diff
@@ -89,16 +99,36 @@ func (i *InputDefinition) LoadDefinition(pb *internal.InputDefinition) error {
 		i.frames = append(i.frames, inputFrame)
 	}
 
+	numPrimaryKey := 0
+	countRowID := make(map[string]uint64)
 	for _, field := range pb.Fields {
 		var actions []Action
 		for _, action := range field.Actions {
+			if err := i.ValidateAction(action); err != nil {
+				return err
+			}
+			if action.RowID != 0 && action.Frame != "" {
```

Comment:
Instead of checking for rowID !=0
if action.ValueDestination == "single-row-boolean"  && action.Frame != "" {